### ISSUE TITLE
[nerdctl] cannot pull\push images with custom "credHelpers"

### ISSUE THREAD

**Issue by Sebor** [rancher-sandbox/rancher-desktop#1367]
### Rancher Desktop Version

1.0.0

### Rancher Desktop K8s Version

1.22.5

### What operating system are you using?

macOS

### Operating System / Build Version

Monterey 12.1

### What CPU architecture are you using?

x64

### Actual Behavior

Cannot pull/push images from/to yandex container images (an analog of AWS ECR).
`credHelpers` in `config.json` file ignores.

### Steps to Reproduce

```bash
nerdctl pull cr.yandex/<registry_id>/app:v1.0.0
```

### Result

```
FATA[0001] failed to resolve reference "cr.yandex/<registry_id>/app:v1.0.0": pulling from host cr.yandex failed with status code [manifests master]: 401 Unauthorized
```

### Expected Behavior

I can pull\push images without errors

### Additional Information

It works with docker-desktop:
```bash
$ docker push cr.yandex/<registry_id>/app:v1.0.0
The push refers to repository [cr.yandex/<registry_id>/app]
c2708a6bd19c: Pushed 
930b18e2bd71: Pushed 
8d3ac3489996: Pushed 
master: digest: sha256:35cce2c4823dfdg34r4gfdg3434g4785a0c25 size: 946
```

My `~/.docker/config.json`:
```json
{
  "credHelpers": {
    "container-registry.cloud.yandex.net": "yc",
    "cr.cloud.yandex.net": "yc",
    "cr.yandex": "yc"
  },
  "credsStore": "desktop"
}
```

**Comment by jdolitsky** [comment]
Seems related: https://github.com/containerd/nerdctl/issues/429#issuecomment-983082580

**Comment by jandubois** [comment]
The problem is that `nerdctl` is just a small shell wrapper; the tool has to run inside the VM to have access to the containerd filesystem. So the credential helpers must be installed and configured inside the VM.

Put the credential helpers into `/usr/local/bin`, and the config into `/root/.docker/config.json`.

You can create a shell inside the VM with

```console
LIMA_HOME="$HOME/Library/Application Support/rancher-desktop/lima" "/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl" shell 0
```

**Comment by jandubois** [comment]
> Seems related: [containerd/nerdctl#429 (comment)](https://github.com/containerd/nerdctl/issues/429#issuecomment-983082580)

Yes, thank you, seems like the same root cause: you need to configure it inside the VM.

**Comment by cmoulliard** [comment]
> you need to configure it inside the VM.

Will the step `how to ssh to the rancher VM` be documented as part of the rancher desktop documentation ?

**Comment by cmoulliard** [comment]
I cannot ssh using the command suggested 
```
LIMA_HOME="$HOME/Library/ApplicatiApplication\ Support/rancher-desktop/lima"
limactl shell 0
FATA[0000] instance "0" does not exist, run `limactl start 0` to create a new instance
```

**Comment by Sebor** [comment]
Hi @jdolitsky, @jandubois!
Thank you for clarification. But I agree with @cmoulliard about documentation - it should be as a part of this one

**Comment by Sebor** [comment]
> export LIMA_HOME="$HOME/Library/Application\ Support/rancher-desktop/lima"
> /Applications/Rancher\ Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl shell 0
> FATA[0000] instance "0" does not exist, run `limactl start 0` to create a new instance

@cmoulliard have you started rancher-desktop?
I got the same error but forgot to start RD

**Comment by cmoulliard** [comment]
> have you started rancher-desktop?

Yes as I see within the menu bar `kubernetes is running` ;-)

**Comment by Sebor** [comment]
@cmoulliard no idea in that case.
I've got another issue: @jandubois, is there any persistent solution? Because after restart RD all changes inside lima are disappeared

**Comment by Sebor** [comment]
@jandubois sorry for my annoying questions, but it doesn't work.
I get the error:
```bash
$ nerdctl push cr.yandex/<registry_id>/demo-app:master
INFO[0000] pushing as a reduced-platform image (application/vnd.docker.distribution.manifest.v2+json, sha256:<>)
WARN[0000] cannot get auth config for authConfigHostname="cr.yandex" (refHostname="cr.yandex")  error="error getting credentials - err: exec: \"docker-credential-yc\": executable file not found in $PATH, out: ``"
...
elapsed: 1.0 s                                                                    total:   0.0 B (0.0 B/s)
FATA[0001] unexpected status: 401 Unauthorized
```

My `override.yaml`:
```yaml
env:
  K3S_EXEC: --disable=traefik
provision:
  - mode: system
    script: |
      #!/bin/sh
      apk add curl bash
      YC_SDK_STORAGE_URL="${YC_SDK_STORAGE_URL:-"https://storage.yandexcloud.net/yandexcloud-yc"}"
      YC_VERSION="${YC_VERSION:-$(curl -s "${YC_SDK_STORAGE_URL}/release/stable")}"
      wget "${YC_SDK_STORAGE_URL}/release/${YC_VERSION}/linux/amd64/yc" -P /usr/local/bin/
      chmod +x /usr/local/bin/yc
      mkdir -p /root/.docker
      cat <<'EOF' > /root/.docker/config.json
      {
        "credHelpers": {
          "container-registry.cloud.yandex.net": "yc",
          "cr.cloud.yandex.net": "yc",
          "cr.yandex": "yc"
        }
      }
      EOF
```
I've checked inside VM:
```bash
$ which yc
/usr/local/bin/y

$ yc -h
Command line interface helps you interact with Yandex.Cloud services

Usage:
  yc <group|command>
...
```

I've even tried to copy my YC credentials to `/root/.config/yandex-cloud/` but no result

**Comment by jandubois** [comment]
Since v1.4.1, Rancher Desktop bundles Docker credential helpers and runs a host-side credential server that reads `credHelpers` entries from `~/.docker/config.json`. Custom helpers installed on your host PATH (including `~/.rd/bin`) work without provisioning anything inside the VM. The `override.yaml` workaround from the comments is no longer necessary. If a vendor-specific helper like `docker-credential-yc` is not bundled, install it on your host and configure it via `credHelpers` — it should work. Please reopen if this does not cover your use case on a current release.
